### PR TITLE
gl_engine: correct the ColorSpace when render Image

### PR DESCRIPTION
**Pull request by RuiwenTang**
Need handle ColorSpace correct when prepare Image texture. If premultiplied is not marked. Needs to change the ColorSpace.

related issue: #2863

### CODE REVIEW

File: src/renderer/gl_engine/tvgGlRenderer.cpp [hunk 1]
```diff
@@ -1281,6 +1281,10 @@ RenderData GlRenderer::prepare(RenderSurface* image, RenderData data, const Matr
         sdata->texId = _genTexture(image);
         sdata->opacity = opacity;
         sdata->texColorSpace = image->cs;
+        if (!image->premultiplied) {
+            if (image->cs == ColorSpace::ABGR8888) sdata->texColorSpace = ColorSpace::ABGR8888S;
```
Comment (hermet) [inline src/renderer/gl_engine/tvgGlRenderer.cpp:1285]: Hmm... I think the image loader might need to assign the proper colorspace value.

**Comment by hermet** [comment]
Quick test, and a bit confusing since it looks fine to me on linux.

image

**Comment by RuiwenTang** [comment]
> Quick test, and a bit confusing since it looks fine to me on linux.
> 
> image

I can only reproduce this issue on My old laptop. Is it caused by different implementations of the png library?

**Comment by hermet** [comment]
@RuiwenTang could you please confirm that fix the issue? https://github.com/thorvg/thorvg/pull/2883

**Pull request closed without merging**